src/main/java/com/google/codeu/servlets/StatsPageServlet.java

```diff
@@ -1,41 +1,128 @@
-package com.google.codeu.servlets;
-
-import java.io.IOException;
-
-import javax.servlet.annotation.WebServlet;
-import javax.servlet.http.HttpServlet;
-import javax.servlet.http.HttpServletRequest;
-import javax.servlet.http.HttpServletResponse;
-
-import com.google.codeu.data.Datastore;
-import com.google.gson.JsonObject;
-
-/**
- * Handles fetching site statistics.
- */
-@WebServlet("/stats")
-public class StatsPageServlet extends HttpServlet{
-
-  private Datastore datastore;
-
-  @Override
-  public void init() {
-    datastore = new Datastore();
-  }
-
-  /**
-   * Responds with site statistics in JSON.
-   */
-  @Override
-  public void doGet(HttpServletRequest request, HttpServletResponse response)
-      throws IOException {
-
-    response.setContentType("application/json");
-
-    int messageCount = datastore.getTotalMessageCount();
-
-    JsonObject jsonObject = new JsonObject();
-    jsonObject.addProperty("messageCount", messageCount);
-    response.getOutputStream().println(jsonObject.toString());
-  }
-}
\ No newline at end of file
+package com.google.codeu.servlets;
+
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.Collection;
+import java.util.Collections;
+import java.util.List;
+import java.util.AbstractCollection;
+
+import javax.servlet.annotation.WebServlet;
+import javax.servlet.http.HttpServlet;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+
+import com.google.appengine.api.mail.MailService.Message;
+import com.google.codeu.data.Datastore;
+import com.google.gson.JsonObject;
+
+/**
+ * Handles fetching site statistics.
+ */
+@WebServlet("/stats")
+public class StatsPageServlet extends HttpServlet{
+
+  private Datastore datastore;
+  @Override
+  public void init() {
+    datastore = new Datastore();
+  }
+
+  /**
+   * Responds with site statistics in JSON.
+   */
+  @Override
+  public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
+
+    response.setContentType("application/json");
+    
+    com.google.codeu.data.Message message = new com.google.codeu.data.Message("Alan", "Ola k ase");

```

Comment:
Please remove these 3 fake messages since the appear to only be used for your testing purposes.